Userland/Libraries/LibJS/Runtime/ArrayConstructor.cpp

```diff
@@ -105,6 +105,12 @@ JS_DEFINE_NATIVE_FUNCTION(ArrayConstructor::from)
         map_fn = &callback.as_function();
     }
 
+    // thisArg argument
+    auto this_value = vm.argument(2);
+    if (this_value.is_undefined()) {
+        this_value = global_object.value_of();
+    }
```

Comment:
This is not part of the spec... they just pass whatever the third argument is directly to the callback.

Also, `this_value` is usually reserved for the actual this value within the function (`vm.this_value()`), so let's call it `this_arg` just like the spec.

```suggestion
    auto this_arg = vm.argument(2);
```

Reply:
tbh I don't think that's correct...
If `thisArg` is not passed or passed as `undefined` the most reasonable choice is to pass `global_object` to `vm.call`.

With your suggestion, we would always give `vm.call` `undefined` should `thisArg` be omitted. And this will trigger `TypeError: ToObject on null or undefined` each time function tries to address `this`

Consider these examples:

should give [ NaN, NaN, NaN ], but will be TypeError
```
Array.from([1, 2, 3], function (n, i) { return n + this.e });   
```

should give [ 101, 102, 103 ], but will be TypeError
```
e = 100;
Array.from([1, 2, 3], function (n, i) { return n + this.e });
```

should give [ 101, 102, 103 ], but will be TypeError
```
e = 100;
Array.from([1, 2, 3], function (n, i) { return n + this.e }, undefined);
```

Reply:
> tbh I don't think that's correct...

It is, see spec. :)

>  If thisArg is not passed or passed as undefined the most reasonable choice is to pass global_object to vm.call.

No.

> With your suggestion, we would always give vm.call undefined should thisArg be omitted.

Yes.

> And this will trigger TypeError: ToObject on null or undefined each time function tries to address this

Then that's a bug in out call implementation. The relevant part of the spec for that is here: https://tc39.es/ecma262/#sec-ordinarycallbindthis
Which gets called by https://tc39.es/ecma262/#sec-ecmascript-function-objects-call-thisargument-argumentslist (step 6). You'll see it has behaviour depending on whether we're currently in strict mode or not - `"use strict"; Array.from([1, 2, 3], function () { console.log(this) });` will print `undefined` three times (in compliant engines, that is), but not if we explicitly pass in the global object.

Reply:
fair enough. You convinced me :) 
I made appropriate changes as well as changes to the PR message.